config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurerTests.java

```diff
@@ -1052,6 +1054,7 @@ SecurityFilterChain chain(HttpSecurity http) throws Exception {
 				.httpBasic(withDefaults())
 				.authorizeHttpRequests((requests) -> requests
 					.requestMatchers("/user/{username}").access(new WebExpressionAuthorizationManager("#username == 'user'"))
+					.requestMatchers("/v2/user/{username}").hasVariable("username").equalTo("user")
```

Comment:
I think a more realistic usage is:

```java
.requestMatchers("/v2/user/{username}").hasVariable("username").equalTo(Authentication::getName)
```

Can we use something like that instead?